storage/file.go

```diff
@@ -53,12 +101,83 @@ type ListSharesParameters struct {
 	Timeout    uint
 }
 
+// DirsAndFilesListResponse contains the response fields from
+// a List Files and Directories call.
+//
+// See https://msdn.microsoft.com/en-us/library/azure/dn166980.aspx
+type DirsAndFilesListResponse struct {
+	XMLName     xml.Name    `xml:"EnumerationResults"`
+	Xmlns       string      `xml:"xmlns,attr"`
+	Marker      string      `xml:"Marker"`
+	MaxResults  int64       `xml:"MaxResults"`
+	Directories []Directory `xml:"Entries>Directory"`
+	Files       []File      `xml:"Entries>File"`
+	NextMarker  string      `xml:"NextMarker"`
+}
+
+// FileRanges contains a list of file range information for a file.
+//
+// See https://msdn.microsoft.com/en-us/library/azure/dn166984.aspx
+type FileRanges struct {
+	ContentLength uint64
+	ETag          string
+	FileRanges    []FileRange `xml:"Range"`
+}
+
+// FileRange contains range information for a file.
+//
+// See https://msdn.microsoft.com/en-us/library/azure/dn166984.aspx
+type FileRange struct {
+	Start uint64 `xml:"Start"`
+	End   uint64 `xml:"End"`
+}
+
+// ListDirsAndFilesParameters defines the set of customizable parameters to
+// make a List Files and Directories call.
+//
+// See https://msdn.microsoft.com/en-us/library/azure/dn166980.aspx
+type ListDirsAndFilesParameters struct {
+	Marker     string
+	MaxResults uint
+	Timeout    uint
+}
+
 // ShareHeaders contains various properties of a file and is an entry
 // in SetShareProperties
 type ShareHeaders struct {
 	Quota string `header:"x-ms-share-quota"`
 }
 
+type compType string
+type httpVerb string
+type resourceType string
+
+const (
+	compNone          compType     = ""
+	compList          compType     = "list"
+	compMetadata      compType     = "metadata"
+	compProperties    compType     = "properties"
+	compRangeList     compType     = "rangelist"
+	httpGet           httpVerb     = "GET"
```

Comment:
Refactor! yesss! I bring [this](https://golang.org/pkg/net/http/#pkg-constants) to the table.